fix: functioning search and multiselect

ColinBuyck opened the pull request:
This PR addresses #4795 

- [x] Addresses the issue in full
- [ ] Addresses only certain aspects of the issue

## Description

This PR does three primary things:

1. Add search by name functionality
2. Add filtering by community type
3. Add unit testing and integration tests
<img width="1431" alt="Screenshot 2025-06-06 at 10 37 57 AM" src="https://github.com/user-attachments/assets/6de95dcd-145a-4622-9c46-64e99f8be0a7" />


## How Can This Be Tested/Reviewed?

This can be tested locally by reseeding and testing the two new filters. Update the listing data and see the filtering results reflect those updates. Also a code walkthrough of the tests to see if any gaps come up would be helpful!

## Author Checklist:

- [x] Added QA notes to the issue with applicable URLs
- [ ] Reviewed in a desktop view
- [ ] Reviewed in a mobile view
- [ ] Reviewed considering accessibility
- [x] Added tests covering the changes
- [ ] Made corresponding changes to the documentation
- [ ] Ran `yarn generate:client` and/or created a migration when required

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates

ColinBuyck commented on sites/public/src/components/browse/FilterDrawerHelpers.tsx:
```diff
@@ -66,19 +73,14 @@ const arrayFilters: ListingFilterKeys[] = [
   ListingFilterKeys.section8Acceptance,
 ]
 
-// https://github.com/bloom-housing/bloom/issues/4795
```
**note:** Captured here: https://github.com/bloom-housing/bloom/issues/4441?issue=bloom-housing%7Cbloom%7C4938

emilyjablonski commented on sites/public/__tests__/components/browse/FilterDrawerHelpers.test.tsx:
```diff
@@ -13,87 +102,751 @@ describe("filter drawer helpers", () => {
+        },
+      ])
+    })
+    it("should return correct fields when existindData is partial form object", () => {
```
https://github.com/bloom-housing/bloom?tab=readme-ov-file#vscode-extensions spell checker plugin here could be helpful!

emilyjablonski commented on sites/public/__tests__/components/browse/FilterDrawerHelpers.test.tsx:
```diff
@@ -13,87 +102,751 @@ describe("filter drawer helpers", () => {
+      )
+    }
+
+    const CustomnColumnCheckBoxGroup = () => {
```
**nitpick:** 
```suggestion
    const CustomColumnCheckBoxGroup = () => {
```

emilyjablonski reviewed (approved):
This LGTM. Just confirming that if the swap community types toggle is off, we're not supposed to see the reserved community types instead? The ticket indicates that is the case, but I feel like we spoke offline and determined this should only do the community type work and didn't update the ticket. Regardless, I would suggest that be future work anyway since we won't release that version soon.

ColinBuyck commented on sites/public/__tests__/components/browse/FilterDrawer.test.tsx:
```diff
@@ -1,3 +1,197 @@
+    expect(screen.getByLabelText("Only show listings confirmed by property"))
+    expect(
+      screen.getByRole("checkbox", { name: "Only show listings confirmed by property" })
+    ).not.toBeChecked()
```
Done!

ColinBuyck commented:
@ludtkemorgan Thanks for the testing advice! Added the tests you recommended!

@emilyjablonski You are remembering correctly that we determined that wasn't desired work right now. I updated the linked ticket and still made an optional ticket stub if we end up revisiting the epic and wanted to add the functionality https://github.com/bloom-housing/bloom/issues/4996

Going to merge once tests pass 🦺
